docs/handcraft.rst

```diff
@@ -0,0 +1,2049 @@
+the lock is basically a public key, such that in order to unlock the
+transaction one needs to have the private key.
+
+Example of a condition payload:
+
+.. ipython::
+
+    {
+        'amount': 1,
+        'cid': 0,
+        'condition': {
+            'details': {
+                'bitmask': 32,
+                'public_key': '8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc',
+                'signature': None,
+                'type': 'fulfillment',
+                'type_id': 4,
+            },
+            'uri': 'cc:4:20:bOZjTedaOgPsbYjh3QeOEQCj1o1lIvVefR71sS8egnM:96'
+        },
+        'owners_after': ['8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc'],
+    }
+    
+
+fulfillments
+^^^^^^^^^^^^
+At the moment, we have to distinguish between fulfillment payloads **before**
+fulfillment and **after** fulfillment. For signature-based schemes, this means,
+before signature(s), and after signature(s).
+
+Moreover payloads for ``CREATE`` operations are a bit different.
+
+.. note:: We hope to be able to simplify the payload strucutre and validation,
+    such that this is no longer require.
+
+.. todo:: Point to issues addressing the topic.
+
+Example of a fulfillment payload **before fulfilling it**, for a `CREATE`
+operation:
+
+.. ipython::
+
+    fulfillment = {
+        'fid': 0,
+        'fulfillment': None,
+        'input': None,
+        'owners_before': ['8L6ngTZ5ixuFEr1GiunrFNWtGkft4swWWArXjWJu2Uwc'],
+    }
+
+.. note:: Because it is a `CREATE` operation, the ``'input'`` field is set to
+    ``None``.
+
+.. todo:: Example of a fulfillment payload **after fulfilling it**:
+
+
+.. _bicycle-asset-creation-revisited:
+
+********************************
+Bicycle Asset Creation Revisited
+********************************
+
+The Prepared Transaction
+========================
+Recall that to prepare a transaction we had to do something similar to:
+
+.. ipython::
+
+    In [0]: from bigchaindb_driver.crypto import generate_keypair
+
+    In [0]: from bigchaindb_driver.offchain import prepare_transaction
+
+    In [0]: alice = generate_keypair()
+
+    In [0]: bicycle = {
+       ...:     'data': {
+       ...:         'bicycle': {
+       ...:             'serial_number': 'abcd1234',
+       ...:             'manufacturer': 'bkfab',
+       ...:         },
+       ...:     },
+       ...: }
+
+    In [0]: metadata = {'planet': 'earth'}
+
+    In [0]: prepared_creation_tx = prepare_transaction(
+       ...:     operation='CREATE',
+       ...:     owners_before=alice.verifying_key,
+       ...:     asset=bicycle,
+       ...:     metadata=metadata,
+       ...: )
+
+and the payload of the prepared transaction looked similar to:
+
+.. ipython::
+    
+    In [0]: prepared_creation_tx
+
+Note ``alice``'s publci key:
+
+.. ipython::
+
+    In [0]: alice.verifying_key
+
+We are now going to craft this payload by hand.
+
+Extract asset id and metadata id:
+
+.. ipython::
+
+    In [0]: asset_id = prepared_creation_tx['transaction']['asset']['id']
```

Comment:
instead of doing this , maybe `str(uuid.uuid4())`.

Unless you want to end up with the exact same payload which is fine, but then just add a note about it. Or else it feels like cheating first creating a transaction with the driver and then using some of its values